registry/storage/driver/s3-aws/s3.go

```diff
@@ -874,6 +875,139 @@ func (d *driver) URLFor(ctx context.Context, path string, options map[string]int
 	return req.Presign(expiresIn)
 }
 
+// Walk traverses a filesystem defined within driver, starting
+// from the given path, calling f on each file
+func (d *driver) Walk(ctx context.Context, from string, f storagedriver.WalkFn) error {
+	var objectCount int64
```

Comment:
Perhaps its better to declare this closer to where it's used, i.e. near the invocation of `doWalk`.